### PR TITLE
Fixes #5545

### LINKED ISSUE #5545: DROP SUBSCRIPTION query called on invalid retention policy corrupts database
It seems like I accidentally corrupted the database by running a `DROP subscription` query while playing around with kapacitor.

- CentOS release 6.6 (Final), 2.6.32-504.16.2.el6.x86_64
- InfluxDB version 0.9.6.1-1 installed from yum

We installed kapacitor and started it with this config.

    [influxdb]
      # Connect to an InfluxDB cluster
      # Kapacitor can subscribe, query and write to this cluster.
      # Using InfluxDB is not required and can be disabled.
      enabled = true
      urls = ["http://localhost:8086"]
      username = "telegraf"
      password = "<%= scope.lookupvar('influx::params::telegraf_password') %>"
      timeout = 0
      # Subscriptions use the UDP network protocl.
      # The following options of for the created UDP listeners for each subscription.
      # Number of packets to buffer when reading packets off the socket.
      udp-buffer = 1000
      # The size in bytes of the OS read buffer for the UDP socket.
      # A value of 0 indicates use the OS default.
      udp-read-buffer = 0

      [influxdb.subscriptions]
        # Set of databases and retention policies to subscribe to.
        # If empty will subscribe to all, minus the list in
        # influxdb.excluded-subscriptions
        #
        # Format
        # db_name = <list of retention policies>
        #
        # Example:
        # my_database = [ "default", "longterm" ]
      [influxdb.excluded-subscriptions]
        # Set of databases and retention policies to exclude from the subscriptions.
        # If influxdb.subscriptions is empty it will subscribe to all
        # except databases listed here.
        #
        # Format
        # db_name = <list of retention policies>
        #
        # Example:
        # my_database = [ "default", "longterm" ]

Since we didn't specify any subscriptions, kapacitor set up the following:

    curl -G http://localhost:8086/query --data-urlencode "q=SHOW subscriptions"

    name: _internal
    ---------------
    retention_policy    name        mode    destinations
    monitor         kapacitor   ANY [udp://localhost:35359]

    name: telegraf
    --------------
    retention_policy    name        mode    destinations
    default         kapacitor   ANY [udp://localhost:50576]

I changed the kapacitor config to only have the subscription `telegraf = ["default"]`, and figured I should delete the old subscription on `_internal`:

    curl -G http://localhost:8086/query --data-urlencode "q=drop subscription kapacitor on _internal.default"

That produced the following output:

    ERR: error parsing query: found DEFAULT, expected identifier at line 1, char 42

So I found this documentation (https://github.com/influxdata/influxdb/blob/master/influxql/INFLUXQL.md) and added quotes to it:

    curl -G http://localhost:8086/query --data-urlencode 'q=drop subscription kapacitor on "_internal"."default"'

Output:

    ERR: Get http://localhost:8086/query?db=_internal&epoch=ns&q=drop+subscription+kapacitor+on+%22_internal%22.%22default%22: EOF

I didn't care too much about the output, thinking I still had my drop syntax wrong, but when I ran `select * from _internal.monitor` I got this:

    ERR: Get http://localhost:8086/query?db=_internal&epoch=ns&q=select+%2A+from+_internal.monitor: dial tcp 127.0.0.1:8086: connection refused

It turns out influxdb died. Here's what the logs said: https://gist.github.com/robinsk/ad7564ddeb469fdb69e7#file-influxdb-log

Mainly:

    [query] 2016/02/05 09:59:22 DROP SUBSCRIPTION kapacitor ON _internal."default"
    panic: runtime error: invalid memory address or nil pointer dereference

When trying to start the influxdb service again, it fails, with almost the same things in the logs:
https://gist.github.com/robinsk/ad7564ddeb469fdb69e7#file-influxdb_starting-log

Is there any way to restore normal operations on this database, or do I have to restore from backup? This is in a vm in vagrant for testing, so it didn't matter _that_ much this time, but that makes me reluctant of running the same in production.

### PR DESCRIPTION
**Pull request by e-dard**
Refactor out the `SUBSCRIPTION` tests for some better coverage and clarity. Ensure that a retention policy error will be returned if you attempt to drop a subscription on a policy that doesn't exist.

Removed `meta.ErrRetentionPolicyNotFound` since everything else uses `influxdb. ErrRetentionPolicyNotFound()`

### CODE REVIEW

**Comment by seiflotfy** [comment]
Tested and it works :+1:

File: services/meta/data.go [hunk 1]
```diff
@@ -574,8 +574,7 @@ func (data *Data) CreateSubscription(database, rp, name, mode string, destinatio
 	rpi, err := data.RetentionPolicy(database, rp)
 	if err != nil {
 		return err
-	}
-	if rpi == nil {
+	} else if rpi == nil {
```
Comment (corylanou) [inline services/meta/data.go]: nit: I think `golint` prefers not having the else as it was previously written.

**Comment by corylanou** [comment]
minor nit (that I'm not 100% sure on).  +1

**Comment by e-dard** [comment]
@corylanou `golint` seems fine with it, though there were some pre-existing lint issues in files I touched in this PR, so I've tidied those up.

**Comment by corylanou** [comment]
+1

**Comment by jwilder** [comment]
:+1:

**Pull request merged**